azure-inventory/lambda/pull_organization_data.py

```diff
@@ -0,0 +1,60 @@
+import boto3
+from botocore.exceptions import ClientError
+
+from azure_lib.common import *
+
+import json
+import os
+import time
+
+import logging
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+logging.getLogger('botocore').setLevel(logging.WARNING)
+logging.getLogger('boto3').setLevel(logging.WARNING)
+
+
+# Lambda main routine
+def handler(event, context):
+    logger.info("Received event: " + json.dumps(event, sort_keys=True))
+
+    dynamodb = boto3.resource('dynamodb')
+    subscription_table = dynamodb.Table(os.environ['SUBSCRIPTION_TABLE'])
+
+    credential_info = get_azure_creds(os.environ['AZURE_SECRET_NAME'])
+    if credential_info is None:
+        raise Exception("Unable to extract Azure Credentials. Aborting...")
+
+    subscription_list = get_subcriptions(credential_info)
+    if subscription_list is None:
+        raise Exception("No Subscriptions found. Aborting...")
+
+    # # print(subscription_list)
+    # for p in subscription_list:
+    #     logger.info(p)
+
+    for p in subscription_list:
+        create_or_update_subscription(p, subscription_table)
+
+    event['subscription_list'] = subscription_list
+    return(event)
+
+# end handler()
+
+##############################################
+
+
+def create_or_update_subscription(subscription, subscription_table):
+    logger.info(u"Adding subscription {}".format(subscription))
+
+    try:
+        response = subscription_table.put_item(Item=subscription)
```

Comment:
This will overwrite any custom attributes applied to the subscription record. Please implement using update_item with just the elements returned from the Azure API.